db/db_impl/db_impl.cc

```diff
@@ -2402,6 +2417,7 @@ Status DBImpl::DropColumnFamilyImpl(ColumnFamilyHandle* column_family) {
                                     mutable_cf_options->max_write_buffer_number;
     }
 
+    bool cf_support_snapshot = cfd->mem()->IsSnapshotSupported();
```

Comment:
Is the logic related to dropping column family removed?
I know it might be challenging to handle drop column family, but it might need to be handled.

Reply:
Yes, I'm still thinking about how to only remove the uniquely owned cf_paths.

Reply:
`DropColumnFamilyImpl` might not be the most appropriate place to call `OnDbPathsRemoved`, because even when CF is dropped, as long as the user hasn't deleted `ColumnFamilyHandle`, the user might still be reading data from the CF through the handle.

Reply:
I think `~ColumnFamilyData` is a better place to call `OnDbPathsRemoved`, because when `~ColumnFamilyData` is called, it means there is no `ColumnFamilyHandle` referencing to the CFD, so there will be no user reading from the SST files in the CF's cf_paths any more.

Reply:
That's a good point.

Reply:
With the new reference counting mechanism in the cache, each column family handle will increase the reference count, so unregister is changed to be called in `~ColumnFamilyHandle` instead of `~ColumnFamilyData` .